// Copyright 2025 The Chromium Authors
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

syntax = "proto3";

package proto_to_value;

option optimize_for = LITE_RUNTIME;

import "components/safe_browsing/core/common/proto_to_value/test_proto/test_proto_dependency.proto";

message TestMessage {
  double double_field = 1;

  int32 int32_field = 2;

  message NestedMessage {
    int32 int32_field = 1;
  }

  NestedMessage nested_message_field = 3;

  enum TestEnum {
    UNKNOWN = 0;
    ENUM_A = 1;
    ENUM_B = 2;
  }

  TestEnum enum_field = 4;

  repeated int32 repeated_int32_field = 5;

  string string_field = 6;

  bytes bytes_field = 7;

  DependencyMessage dependency_message = 8;
}
